app/concepcts/reservation_seats/validator.rb

```diff
@@ -0,0 +1,19 @@
+module ReservationSeats
+  class Validator
+    def initialize(seat_ids, screening_id, repository: ReservationSeats::Repository.new)
+      @seat_ids = seat_ids
+      @screening_id = screening_id
+      @repository = repository
+    end
+    #potrzebuje znalezc  ilsoc miejsc jaka jest całkowiat dla danego screeningu, mogę to uzystac ze screeningu, albo przekazujac ille jest miejsc dla danesj cinema hall 
+    def valid?
+      seats_reserved = repository.reserved_seats(screening_id)
+      all_seats = repository.all(screening_id)
+      (seats_reserved & seat_ids).any? or (all_seats.count - seats_reserved.count > seat_ids.count)
```

Comment:
`(seats_reserved & seat_ids).any?` means that "if there are overlapping seats return true"
It's valid approach when you returning information if something is valid?

Also

```(seats_reserved & seat_ids).any? OR (all_seats.count - seats_reserved.count > seat_ids.count)```
Do you think that only one from those conditionals has to be true to return that everything is valid?

Reply:
Approach with any is not valid. Now I am using none   `seat_ids.none? { |id| seats_reserved.include?(id) }`. 
Also change that  two of those condition have to be valid `seats_free(seats_reserved) and enough_seats?(all_seats, seats_reserved)`.